src/astro/databases/google/bigquery.py

```diff
@@ -135,3 +146,100 @@ def merge_table(
             )
             statement += f" WHEN MATCHED THEN {update_statement}"
         self.run_sql(sql_statement=statement)
+
+    def check_optimised_path(self, source_file: File, target_table: Table) -> bool:
+        """
+        Check if there is an optimised path for source to destination.
+
+        :param source_file: File from which we need to transfer data
+        :param target_table: Table that needs to be populated with file data
+        """
+        return bool(self.OPTIMIZED_PATHS.get(source_file.location.location_type))
+
+    def optimised_transfer(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Checks if optimised path for transfer between File location to database exists
+        and if it does, it transfers it and returns true else false.
+        """
+        method_name = self.OPTIMIZED_PATHS.get(source_file.location.location_type)
+        if method_name:
+            transfer_method = self.__getattribute__(method_name)
+            transfer_method(
+                source_file=source_file,
+                target_table=target_table,
+                if_exists=if_exists,
+                **kwargs,
+            )
+
+    @staticmethod
+    def get_project_id_from_conn(conn: connection) -> str:
+        """
+        Get project id from conn
+
+        :param conn: Airflow's connection
+        """
+        if conn.extra and conn.extra_dejson.get("project"):
+            return str(conn.extra_dejson["project"])
+        elif conn.host:
+            return str(conn.host)
+        raise ValueError(f"conn_id {conn.conn_id} has no project id.")
+
+    def gs_to_bigquery(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ) -> None:
+        """
+        Transfer data from gcs to bigquery
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        bq_hook = BigQueryHook(
+            gcp_conn_id=target_table.conn_id,
+        )
+
+        if if_exists == "replace":
+            self.drop_table(target_table)
+
+        file_type_to_bq_source_format = {
+            FileType.CSV: "CSV",
+            FileType.NDJSON: "NEWLINE_DELIMITED_JSON",
+            FileType.PARQUET: "PARQUET",
```

Comment:
What happens if the user had a `FileType.JSON`? Should we raise an exception?

Reply:
@tatiana @kaxil @sunank200  Nice catch! Yes, we should, and I'll update the code reflecting the same. 

Also, there are two things here
1. I think we should drop the support for the JSON files as they are not used for large data transfer the preferred file type is NDJSON and most DBs actually mean NDJSON when they say JSON.
2. We should also include fileTypes in figuring out optimized paths. Right now we are using only (source, destination) it should be (source, destination, fileType) since there might be a support to ingest some fileType and not others by DBs and we should take that into account.

Reply:
@utkarsharma2:
1. I'm fine dropping support to JSON that is not new-line delimited.
2. I have mixed feelings about this. It would be great if the Astro SDK converts between file types, if necessary, to support the optimised path regardless of the file type.

Reply:
1. I think we discussed on Friday that we will fall-back to the non-optimized path if the filetype isn't supported or raise an error for now and drop the support in future major releases
2. I agree we should include filetype in finding optimized path as doing without it means a) we will either need to fallback to pandas or b) convert the file on the source. So happy for you to include filetype in optimization check. And CSV, NDJSON, PARQUET should be supported by most of the DBs anyways

Reply:
Noted. Made the required code changes.